src/components/project/project.service.ts

```diff
@@ -483,10 +490,13 @@ export class ProjectService {
       changes,
       nodevar: 'project',
     });
-
-    await this.eventBus.publish(
-      new ProjectUpdatedEvent(result, currentProject, input, session)
+    const projectUpdatedEvent = new ProjectUpdatedEvent(
+      result,
+      currentProject,
+      input,
+      session
     );
+    await this.eventBus.publish(projectUpdatedEvent);
 
     return await this.readOne(input.id, session);
```

Comment:
If `sgUpdateProperties` returns the updated object then why are we calling readOne again?

I know the new handler currently needs it because it update the object returned but I'd rather have it modify the event object like we are doing in other places.

If there are other reasons for this extra read call we should fix them too.
If the read call is because the `result` from `sgUpdatedProperties` doesn't have the correct info then we should just ignore it and do the readOne before publishing the event.

Reply:
So is your preference to have these changes reversed: https://github.com/SeedCompany/cord-api-v3/pull/1294/commits/d444fe1572ed72ae8b4a7a023018b3f2a9ec3a88

Reply:
Well kinda. The handler is just changing the `departmentId` so there's no need to call `readOne` again. Just "modify"  (spread the object into a new one) the project object and adjust the one property which you should already have the new value for. Said another way, you already have the fresh data to update the stale parts of the object so there's no need to ask for it again from the db.